src/Components/test/E2ETestMigration/Infrastructure/ComponentTestBase.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.BrowserTesting;
+using Microsoft.AspNetCore.Testing;
+using PlaywrightSharp;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore.Components.E2ETest.Infrastructure
+{
+    public abstract class ComponentBrowserTestBase : BrowserTestBase
+    {
+        public ComponentBrowserTestBase(ITestOutputHelper output = null) : base(output) { }
+
+        protected abstract Type TestComponent { get; }
+        protected string MountUri { get; set; }
+        protected IPage TestPage { get; set; }
+        protected IBrowserContext TestBrowser { get; set; }
+        protected BrowserKind BrowserKind { get; set; }
+
+        protected async Task MountTestComponentAsync(IPage page)
+        {
+            var componentTypeName = TestComponent.Assembly == typeof(BasicTestApp.Program).Assembly ?
+                TestComponent.FullName :
+                TestComponent.AssemblyQualifiedName;
+            var testSelector = await page.WaitForSelectorAsync("#test-selector > select");
+            Assert.NotNull(testSelector);
+
+            Output.WriteLine("Selecting test: " + componentTypeName);
+
+            var selected = await page.SelectOptionAsync("#test-selector > select", componentTypeName);
+            Assert.True(selected.Length == 1);
+            Assert.Equal(componentTypeName, selected.First());
+        }
+
+        public bool ShouldSkip(BrowserKind browserKind) => !BrowserManager.IsAvailable(browserKind);
+
+        protected override async Task InitializeCoreAsync(TestContext context)
+        {
+            await base.InitializeCoreAsync(context);
+
+            // Default to Chrome
+            var browserKind = BrowserKind.Chromium;
+
+            // Check if a different browser is requested
+            var browserKindArgument = context.MethodArguments.FirstOrDefault();
+            if (browserKindArgument != null &&
+                browserKindArgument.GetType() == typeof(BrowserKind))
+            {
+                browserKind = (BrowserKind)browserKindArgument;
+            }
+
+            if (ShouldSkip(browserKind))
+            {
+                return;
+            }
+
+            TestBrowser = await BrowserManager.GetBrowserInstance(browserKind, BrowserContextInfo);
```

Comment:
Is this launching a new browser every time?

Reply:
Yes, I still have to get to https://github.com/dotnet/aspnetcore/issues/31111 which will tackle figuring out the best way to share fixtures/browser instances.  Seems like we could just have a static BrowserManager instead of a shared fixture too right?